runtime/lua/vim/shared.lua

```diff
@@ -347,6 +347,30 @@ function vim.tbl_add_reverse_lookup(o)
   return o
 end
 
+--- Index into a table using a list of keys. Return `nil` if the key does not exist.
+--- Examples:
+--- <pre>
+---  vim.tbl_get({ key = { nested_key = true }}, {'key', 'nested_key'}) == true
+---  vim.tbl_get({ key = {}}, {'key', 'nested_key'}) == nil
+--- </pre>
+---
+---@param o Table to index
+---@param keys List of indices via which to index the table
+---
+---@returns nested value indexed by key if it exists, else nil
+function vim.tbl_get(o, keys)
```

Comment:
What about if you defined keys as varargs (`...`)?

Reply:
The main reason I didn't do this is because I optimized for the case of passing the list retrieved from a given table, such as here: https://github.com/neovim/neovim/pull/17814/files#diff-c6a4160a416b22cd1fd833b8f852c4fbaf1f25adf75053e82a1025aa2c607ac9R35-R54

https://github.com/neovim/neovim/pull/17814/files#diff-c6a4160a416b22cd1fd833b8f852c4fbaf1f25adf75053e82a1025aa2c607ac9R973-R977

Reply:
You mean `vim.tbl_get(client.resolved_capabilities, required_capability)`? That would just become `vim.tbl_get(client.resolved_capabilities, unpack(required_capability))`.

And then the common case `vim.tbl_get(client.resolved_capabilities, {"textDocumentSync", "change"})` becomes `vim.tbl_get(client.resolved_capabilities, "textDocumentSync", "change")`, which I would expect the vast majority of uses would be.

I really don't think either form will have any meaningful performance difference; the most ergonomic should win out as a stdlib addition.

Reply:
Yeah, I just found unpack a bit unasthetic. I'm almost entirely neutral. @gpanders @mfussenegger @muniter wdyt?

Reply:
I think I agree with @lewis6991 actually. The common case will be to pass individual strings, for which varargs is (slightly) more ergonomic. I think the case of having an existing table and needing to use unpack() will be relatively rare.